src/Laravel/Security/ResourceAccessChecker.php

```diff
@@ -13,13 +13,27 @@
 
 namespace ApiPlatform\Laravel\Security;
 
+use ApiPlatform\Laravel\Eloquent\Paginator;
 use ApiPlatform\Metadata\ResourceAccessCheckerInterface;
 use Illuminate\Support\Facades\Gate;
 
 class ResourceAccessChecker implements ResourceAccessCheckerInterface
 {
     public function isGranted(string $resourceClass, string $expression, array $extraVariables = []): bool
     {
-        return Gate::allows($expression, $extraVariables['object']);
+        // ugly way to handle collection
+        if (($object = $extraVariables['object']) instanceof Paginator) {
+            // do not deny access if no items are found
+            if (0 === $object->count()) {
+                return true;
+            }
+
+            // we're only checking access to the first item
+            foreach ($object as $obj) {
+                return Gate::allows($expression, $obj);
+            }
```

Comment:
```suggestion
            // we're only checking access to the first item
            foreach ($object as $obj) {
                return Gate::allows($expression, $obj);
            }
            
            return true;
```

micro-optim (to be tested)

Reply:
I refactored this part, therefore this code is outdated